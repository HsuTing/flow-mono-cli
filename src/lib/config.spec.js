// @flow

const config = require('./config.js');

describe('config.resolveAndReadConfig()', () => {
  let search;

  beforeEach(() => {
    search = jest.fn();
    jest.spyOn(config._utils, 'cosmiconfig').mockImplementation(
      jest.fn(() => ({
        search
      }))
    );
  });

  afterEach(() => {
    // $FlowFixMe: Ignore errors since the jest type-def is out of date.
    jest.restoreAllMocks();
    jest.clearAllMocks();
  });

  it('should be a function', () => {
    expect(typeof config.resolveAndReadConfig).toBe('function');
  });

  it('should call the "find-config-up" package and return the resolved config', async () => {
    search.mockReturnValue({config: {foo: 'bar'}});

    const cfg = await config.resolveAndReadConfig();

    expect(cfg).toMatchSnapshot();
  });
});
