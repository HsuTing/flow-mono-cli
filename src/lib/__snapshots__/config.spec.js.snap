// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config.resolveAndReadConfig() should call the "find-config-up" package and return the resolved config 1`] = `
Object {
  "create-stubs": Object {
    "dependencies": Array [],
  },
  "create-symlinks": Object {
    "ignore": Array [],
  },
  "flowTypedCommandExecRetries": 1,
  "foo": "bar",
}
`;
